xds/src/main/java/io/grpc/xds/sds/SslContextProvider.java

```diff
@@ -27,22 +29,33 @@
  * secret(s) that are dynamic.
  */
 @Internal
-public interface SslContextProvider {
+public abstract class SslContextProvider<K> {
+
+  private final K key;
 
-  interface Callback {
+  public interface Callback {
     /** Informs callee of new/updated SslContext. */
     void updateSecret(SslContext sslContext);
 
     /** Informs callee of an exception that was generated. */
     void onException(Throwable throwable);
   }
 
+  protected SslContextProvider(K key) {
```

Comment:
nit: key -> source? (I commented in other, but i am starting new to not dilute the other thread) 
key is one use of this field, but this class is not tied to the ReferenceCountingMap.